flake.nix

```diff
@@ -190,6 +190,33 @@
           };
         };
 
+        formatter = pkgs.treefmt.withConfig {
+          runtimeInputs = with pkgs; [
+            nixfmt-rfc-style
+            stylish-haskell
+          ];
+
+          settings = {
+            on-unmatched = "info";
+            tree-root-file = "flake.nix";
```

Comment:
Note: in the next treefmt version this won't be needed anymore.

https://github.com/numtide/treefmt/pull/571 made it so that by default the tree root is now found by asking `git` for the repo's root.

Probably not worth waiting for that before merging this, though.